### ISSUE TITLE
DHW Temperature Attribute Not Updating

### ISSUE THREAD

**Issue by jarrah31** [zxdavb/evohome#30]
It appears that the Temperature attribute within water_heater.dhw is no longer being updated when using commit 1be968a as it's always stuck at 54. The value is correctly being updated within the Status attribute, but unfortunately I can't get to this number within InfluxDB and Grafana.

It was working before I upgraded to this version but not sure which commit I was previously on. 

Thanks.

**Comment by zxdavb** [comment]
Bugger - there was a big rewrite recently - I wonder if it was in there somewhere.

**Comment by zxdavb** [comment]
OK... Recently, I switched DHW from being a `climate` entity, to a `water_heater`.  

In HA, `water_heater`s do not have a `current_temperature`, only a `target_temperature`!

This was despite a discussion [here](https://github.com/home-assistant/architecture/issues/59), and I have subsequently complained about that [here](https://github.com/home-assistant/architecture/issues/117).

HA is great, but things can move slowly at times, which is why I've switched from working on the official HA version of the evohome integration, to a custom_component version.

That said, I am reluctant to move too far from HA's view of things.

The `current_temp` of the DHW is still available in `device_state_attributes`, is that not useful to you?  You could pull the data directly, or abstract it via a sensor (which may offer future proofing).

**Comment by zxdavb** [comment]
This may be useful to you:
```
value_template: "{{ state_attr('water_heater.~DHW', 'status').temperatureStatus.temperature }}"
```

Note I haven't tested it, though it is the way to do it.

**Comment by zxdavb** [comment]
BTW...

Why 54?  Well **evohome** does not report the current `target_temperature`, even though it is configurable parameter via the controller, and (presumably) via the web site.

My understanding is that 54 is the default value, and if this is not the case for you, it can changed via a configuration.yaml parameter:
```yaml
evohome_cc:
   dhw_target_temp: 54
```

**Comment by jarrah31** [comment]
ah yes, it is a shame that HA won't adopt current_temperature as it is much more useful to see that number (which varies throughout the day) than a target temperature which never changes. :(

Thank you for supplying the sensor example, it saves me having to figure it out. I can confirm that using the following works well for me (the 2nd states line made more sense to me):
```
sensor:
  - platform: template
    sensors:
      dhw_temp:
        friendly_name: "DHW Temp"
        unit_of_measurement: '°C'
        device_class: temperature
        value_template: >-
          {{ states.water_heater.dhw.attributes['status'].temperatureStatus.temperature }}
```
Using the sensor template is a definite requirement for me for two reasons (might be worth adding the sensor template to the docs):
- As seen in the screenshot below, water_heater.dhw only reports 54 (which I realise is beyond your control as explained above)
- Graphing the values in Grafana via Influxdb means that I'm limited to the available top-level fields, e.g. min_temp, max_temp, temperature, operation_mode, operation_list, etc, so the sensor template is very much needed. 
<img width="483" alt="image" src="https://user-images.githubusercontent.com/3072303/52915783-f4e86200-32cf-11e9-952b-e92cba0fc4c6.png">

Thank you!

**Comment by zxdavb** [comment]
You know, Open Source is a bit like communism: "Each takes as little as they need, and contributes as much as they can".

With that in mind (and remembering all that time I saved you trying to come up with a value template):

https://github.com/zxdavb/evohome/wiki/_new 

(I think just a cut/paste/edit of what is in this thread would do)

**Comment by jarrah31** [comment]
Happy to help contribute to the Wiki (if this is what you meant above?), but I don't have permission to edit it?

**Comment by zxdavb** [comment]
try now?

**Comment by jarrah31** [comment]
Yep, that works now, thanks. 

I'll be able to have a go at updating it this evening.

**Comment by jarrah31** [comment]
Is the data sampling rate lower for water_heater compared to climate out of interest? I've noticed a definite reduction in the detail shown for my DHW graphs compared to how it used to be. The graphs below show the difference between the two. (unfortunately I have no evidence how it used to be because I deleted the DB table for DHW whilst troubleshooting the stuck temp issue)

![image](https://user-images.githubusercontent.com/3072303/53198883-7e23cf80-3615-11e9-9054-0f565123b0c9.png)

btw, happy to give you access to my DHW if it'll help with troubleshooting.
